caching/update.py

```diff
@@ -79,6 +81,33 @@ def main():
         except KeyError:
             builds_that_are_missing_metadata.append(entry)
 
+    # some builds may be missing metadata due to patches failing to apply
+    for build_id in builds_that_are_missing_metadata:
+        build = builds[build_id]
+
+        if "Unable to apply kernel patch" not in build["status_message"]:
+            continue
+        req = urllib.request.Request(build["download_url"] + "build.log")
+        build_log_raw = ""
+        with urllib.request.urlopen(req) as response:
+            build_log_raw = response.read().decode()
+
+        failed_pattern = (
+            r"(?<=Apply patch set FAILED\s)[0-9A-Za-z._:/\-\s]*?(?=\serror: )")
+        failed_matches = re.findall(failed_pattern, build_log_raw)
+        if len(failed_matches) == 0:
+            print(
+                f"No patches failed to apply yet the build status stated there were: {build['status_message']}"
+            )
+            sys.exit(
+                0)  # Not sure how we got here but continue the action anyways
```

Comment:
Yeah, I hate this formatting too. But this is what had to be done to appease the linter gods I guess. I'm pretty sure this is from `$ black caching/update.py`

Reply:
I don't really care when `yapf` or other formatters mess with formatting and make it somewhat wonky. I think moving the comment up a line to fix this is probably worth it since it is going to split it across two lines anyways. Up to you though!